composer.json

```
@@ -28,7 +28,7 @@
     },
     "require-dev": {
         "producer/producer": "^2.3",
-        "phpunit/phpunit": "^9.5",
+        "yoast/phpunit-polyfills": "^1.0",
```

```suggestion
        "yoast/phpunit-polyfills": "^1.0",
        "phpunit/phpunit": "^8.5",
```
This change will prevent the error in the lowest version.

---

Ok. Now I see where it comes from. But does installing the lowest version has any value.

---

May be we can remove the lowest version test

---

The test verifies that the psr/container version specification is correct. (Not for testing tool)

---

`"psr/container": "^1.1 || ^2.0"` passes test but not with `"psr/container": "^1.0"`.

---

I'm not opposed to removing the lower test though.

---

@koriym the error is not due to psr/container version constraints. It is basically as you suggested the phpunit version that is being downloaded is `  - Locking phpunit/phpunit (4.8.36)`. That will anyway fail on 8.x.

---

Yes, it is. The reason why the tests fail is because of the lower version of phpunit.
What I am saying is that the lower test is to check the version of the dependent package.

---

@koriym I added `phpunit/phpunit` to >= 8.0 . This way it can continue to install later versions. Your proposed one will only install 8.5 till 9x. Not 9.x itself. 

Thank you for the help to fix this.